sources/webpki-roots-shim/src/lib.rs

```diff
@@ -29,7 +29,13 @@ lazy_static::lazy_static! {
 fn tls_server_roots_pem() -> Vec<Pem> {
     match std::fs::read(&*CERT_PATH) {
         Ok(data) => {
-            let mut v = pem::parse_many(&data);
+            let mut v = match pem::parse_many(&data) {
+                Ok(v) => v,
+                Err(err) => {
+                    error!("failed to parse {}: {}", CERT_PATH.display(), err);
+                    Vec::new()
```

Comment:
We probably should modify this function to return a `Result` if we can, I'd prefer it fail rather than disregard invalid certs.

Reply:
At the moment we're already disregarding invalid certs further down in the `tls_server_roots_pem` function, so I'm a bit reluctant to change behavior in a dependency update commit. Should I open up an issue to re-evaluate how this handles errors?

Reply:
> reluctant to change behavior in a dependency update commit

I think I agree with your assessment. Previously `parse_many` was swallowing errors, so to remain consistent we should swallow errors here. Otherwise, somebody could have a new Bottlerocket error that they didn't have before with an invalid cert.

Previous error swallowing:

https://github.com/jcreekmore/pem-rs/blob/e869f244552321015e0173a3c71dbb45a43cb981/src/lib.rs#L324